lib/ansible/modules/network/nxos/nxos_vpc.py

```diff
@@ -262,19 +265,41 @@ def get_vpc(module):
                 if 'peer-gateway' in each:
                     vpc['peer_gw'] = False if 'no ' in each else True
                 if 'peer-keepalive destination' in each:
-                    line = each.split()
-                    vpc['pkl_dest'] = line[2]
-                    vpc['pkl_vrf'] = 'management'
-                    if 'source' in each:
-                        vpc['pkl_src'] = line[4]
-                        if 'vrf' in each:
-                            vpc['pkl_vrf'] = line[6]
-                    else:
-                        if 'vrf' in each:
-                            vpc['pkl_vrf'] = line[4]
+                    # destination is reqd; src & vrf are optional
+                    m = re.search(r'destination (?P<pkl_dest>[\d.]+)'
+                                  r'(?:.* source (?P<pkl_src>[\d.]+))*'
+                                  r'(?:.* vrf (?P<pkl_vrf>\S+))*',
+                                  each)
+                    if m:
+                        for pkl in m.groupdict().keys():
+                            if m.group(pkl):
+                                vpc[pkl] = m.group(pkl)
     return vpc
 
 
+def pkl_dependencies(module, delta, existing):
+    """peer-keepalive dependency checking.
+    'destination' is required with all pkl configs.
+    If delta has optional pkl keywords present, then all optional pkl keywords
+    in existing must be added to delta (if specified by the playbook);
+    otherwise the device cli will remove those values.
+    Example CLI:
+      peer-keepalive dest 10.1.1.1 source 10.1.1.2 vrf blue
+
+    {pkl_dest: 10.1.1.1, pkl_src: 10.1.1.2, pkl_vrf: orange} -> CLI changes to vrf orange
+    {pkl_dest: 10.1.1.1, pkl_vrf: blue}                      -> CLI removes source
+    """
+    pkl_existing = [i for i in existing.keys() if i.startswith('pkl')]
+
+    for pkl in pkl_existing:
+        param = module.params.get(pkl)
+        if not delta.get(pkl) and param and param == existing[pkl]:
+            # delta is missing this param because it's idempotent;
+            # however another pkl command has changed; therefore
+            # explicitly add it to delta so that the cli retains it.
+            delta[pkl] = existing[pkl]
+
```

Comment:
Note: Mike wants to preserve "merge" behavior, so always retain existing source even if playbook does not specify.

Reply:
latest commit now does merge.